src/core/env.ml

```diff
@@ -77,3 +77,34 @@ let of_prod_vars : tvar array -> term -> env * term = fun vars t ->
             build_env (i+1) env (Bindlib.subst b (Vari v))
          | _ -> raise (Invalid_argument "of_prod")
   in build_env 0 [] t
+
+(** Given a metavariable [m] of arity [n] and type [∀x1:A1,..,∀xn:An,B] (with
+   [B] being a sort normally), [extend_meta_type m] returns
+   [m[x1,..,xn],(∀y:p,q),bp,bq] where p=m1[x1,..,xn], q=m2[x1,..,xn,y], bp is
+   a mbinder of [x1,..,xn] over p, and bq is a mbinder of [x1,..,xn] over q,
+   where [y] is a fresh variable, and [m1] and [m2] are fresh metavariables of
+   arity [n] and [n+1], and type [∀x1:A1,..,∀xn:An,TYPE] and
+   [∀x1:A1,..,∀xn:An,∀y:m1[x1,..,xn],B] respectively. *)
+let extend_meta_type : meta -> term * term
+    * (term, term) Bindlib.mbinder * (term, tbinder) Bindlib.mbinder
+  = fun m ->
```

Comment:
It would seem that only the two first elements of the result tuple are ever used. I would suggest simplifying this function to only return these two, this way the type will be easier to understand and the doc can be simplified accordingly.

(I'll need to read the doc and code of this function more carefully tomorrow, I'm too tired for that kind of thing now.)

Reply:
No, the two last elements are also used in infer.ml. That's why I introduced this function.

Reply:
Oh I missed that when I used Ctrl+F for some reason. :)